test/kubernetes/e2e/test.go

```diff
@@ -314,6 +320,85 @@ func (i *TestInstallation) PreFailHandler(ctx context.Context) {
 	kubectlGetResourcesCmd := i.Actions.Kubectl().Command(ctx, "get", strings.Join(resourcesToGet, ","), "-A", "-owide")
 	_ = kubectlGetResourcesCmd.WithStdout(clusterStateFile).WithStderr(clusterStateFile).Run()
 	clusterStateFile.WriteString("\n")
+
+	podStdOut := bytes.NewBuffer(nil)
+	podStdErr := bytes.NewBuffer(nil)
+
+	// Fetch the name of the Gloo Gateway controller pod
```

Comment:
Making it easier for customers to get snapshots and other useful debugging information is a good idea. My goal of this PR is to ensure that we/engs can more easily debug test failures in the short-term.

I agree with unifying the code for getting the snapshots between our test frameworks, which if done correctly should make the user face work easier to do later. I will get the framework unification work done in this PR. 

I would like to defer any user facing work to future feature tickets.

Reply:
Makes sense to me! More wanted to point out the existing user-facing work as something we could piggy back off of, since it does some of what you are trying to do here. I'm good with taking the phased approach of enabling devs first, and then we can work to enable users after